File: src/main/java/ch/xxx/manager/entity/Portfolio.java
/**
 *    Copyright 2019 Sven Loesekann
   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at
       http://www.apache.org/licenses/LICENSE-2.0
   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
 */
package ch.xxx.manager.entity;

import java.math.BigDecimal;
import java.time.LocalDate;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;


@Entity
public class Portfolio {
	@Id
	@GeneratedValue(strategy = GenerationType.SEQUENCE)
	private Long id;
	private Long userId;
	private String name;
	private LocalDate createdAt;
	private BigDecimal month1;
	private BigDecimal month6;
	private BigDecimal year1;
	private BigDecimal year2;
	private BigDecimal year5;
	private BigDecimal year10;

	public LocalDate getCreatedAt() {
		return createdAt;
	}
	public void setCreatedAt(LocalDate createdAt) {
		this.createdAt = createdAt;
	}
	public Long getId() {
		return id;
	}
	public void setId(Long id) {
		this.id = id;
	}
	public Long getUserId() {
		return userId;
	}
	public void setUserId(Long userId) {
		this.userId = userId;
	}
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
	public BigDecimal getMonth1() {
		return month1;
	}
	public void setMonth1(BigDecimal month1) {
		this.month1 = month1;
	}
	public BigDecimal getMonth6() {
		return month6;
	}
	public void setMonth6(BigDecimal month6) {
		this.month6 = month6;
	}
	public BigDecimal getYear1() {
		return year1;
	}
	public void setYear1(BigDecimal year1) {
		this.year1 = year1;
	}
	public BigDecimal getYear2() {
		return year2;
	}
	public void setYear2(BigDecimal year2) {
		this.year2 = year2;
	}
	public BigDecimal getYear5() {
		return year5;
	}
	public void setYear5(BigDecimal year5) {
		this.year5 = year5;
	}
	public BigDecimal getYear10() {
		return year10;
	}
	public void setYear10(BigDecimal year10) {
		this.year10 = year10;
	}
	
	@Override
	public String toString() {
		return "PortfolioEntity [id=" + id + ", userId=" + userId + ", name=" + name + ", createdAt=" + createdAt
				+ ", month1=" + month1 + ", month6=" + month6 + ", year1=" + year1 + ", year2=" + year2 + ", year5="
				+ year5 + ", year10=" + year10 + "]";
	}
	
}

File: frontend/src/angular/src/app/model/portfolio.ts
/**
 *    Copyright 2019 Sven Loesekann
   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at
       http://www.apache.org/licenses/LICENSE-2.0
   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
 */
import { Symbol } from "./symbol";
import { PortfolioElement } from "./portfolio-element";

export interface Portfolio extends CommonValues {
  userId: number;
  symbols: Symbol[];
  portfolioElements: PortfolioElement[];
  
  year1CorrelationSp500?: number;
  year1CorrelationMsciChina?: number;
  year1CorrelationEuroStoxx50?: number;
  year1LinRegReturnSp500?: number;
  year1LinRegReturnMsciChina?: number;
  year1LinRegReturnEuroStoxx50?: number;
  year1SigmaSp500?: number;
  year1SigmaMsciChina?: number;
  year1SigmaEuroStoxx50?: number;
  year1SigmaPortfolio?: number;
  year2CorrelationSp500?: number;
  year2CorrelationMsciChina?: number;
  year2CorrelationEuroStoxx50?: number;
  year2LinRegReturnSp500?: number;
  year2LinRegReturnMsciChina?: number;
  year2LinRegReturnEuroStoxx50?: number;
  year2SigmaSp500?: number;
  year2SigmaMsciChina?: number;
  year2SigmaEuroStoxx50?: number;
  year2SigmaPortfolio?: number;
  year5CorrelationSp500?: number;
  year5CorrelationMsciChina?: number;
  year5CorrelationEuroStoxx50?: number;
  year5LinRegReturnSp500?: number;
  year5LinRegReturnMsciChina?: number;
  year5LinRegReturnEuroStoxx50?: number;
  year5SigmaSp500?: number;
  year5SigmaMsciChina?: number;
  year5SigmaEuroStoxx50?: number;
  year5SigmaPortfolio?: number;
  year10CorrelationSp500?: number;
  year10CorrelationMsciChina?: number;
  year10CorrelationEuroStoxx50?: number;
  year10LinRegReturnSp500?: number;
  year10LinRegReturnMsciChina?: number;
  year10LinRegReturnEuroStoxx50?: number;
  year10SigmaSp500?: number;
  year10SigmaMsciChina?: number;
  year10SigmaEuroStoxx50?: number;
  year10SigmaPortfolio?: number;
}

export interface CommonValues {
  id: number;
  currencyKey: string;
  name: string;
  createdAt: string;
  month1: number;
  month6: number;
  year1: number;
  year2: number;
  year5: number;
  year10: number;
}
